sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryPage.scala

```diff
@@ -0,0 +1,148 @@
+
+  def generateDataRow(request: HttpServletRequest, queryActive: Boolean)
+    (streamQuery: (StreamingQuery, Long)): Seq[Node] = {
+
+    val (query, timeSinceStart) = streamQuery
+    def details(detail: Any): Seq[Node] = {
+      val s = detail.asInstanceOf[String]
+      val isMultiline = s.indexOf('\n') >= 0
+      val summary = StringEscapeUtils.escapeHtml4(
+        if (isMultiline) s.substring(0, s.indexOf('\n')) else s
+      )
+      val details = if (isMultiline) {
+        // scalastyle:off
+        <span onclick="this.parentNode.querySelector('.stacktrace-details').classList.toggle('collapsed')"
+              class="expand-details">
+          +details
+        </span> ++
+          <div class="stacktrace-details collapsed">
+            <pre>{s}</pre>
+          </div>
+        // scalastyle:on
+      } else {
+        ""
+      }
+      <td>{summary}{details}</td>
+    }
+
+    val statisticsLink = "%s/%s/statistics?id=%s"
+      .format(SparkUIUtils.prependBaseUri(request, parent.basePath), parent.prefix, query.runId)
+
+    val name = UIUtils.getQueryName(query)
+    val status = UIUtils.getQueryStatus(query)
+    val duration = if (queryActive) {
+      SparkUIUtils.formatDurationVerbose(System.currentTimeMillis() - timeSinceStart)
+    } else {
+      withNoProgress(query, {
+        val endTimeMs = query.lastProgress.timestamp
+        val startTimeMs = query.recentProgress.head.timestamp
+        SparkUIUtils.formatDurationVerbose(
+          df.parse(endTimeMs).getTime - df.parse(startTimeMs).getTime)
+      }, "-")
+    }
+
+    <tr>
+      <td> {name} </td>
+      <td> {status} </td>
+      <td> {query.id} </td>
+      <td> <a href={statisticsLink}> {query.runId} </a> </td>
+      <td> {SparkUIUtils.formatDate(timeSinceStart)} </td>
+      <td> {duration} </td>
+      <td> {withNoProgress(query, {
+        (query.recentProgress.map(p => withNumberInvalid(p.inputRowsPerSecond)).sum /
+          query.recentProgress.length).formatted("%.2f") }, "NaN")}
+      </td>
+      <td> {withNoProgress(query, {
+        (query.recentProgress.map(p => withNumberInvalid(p.processedRowsPerSecond)).sum /
+          query.recentProgress.length).formatted("%.2f") }, "NaN")}
+      </td>
+      <td> {withNoProgress(query,
+        { query.getQuerySummary.getMetric(QuerySummary.TOTAL_INPUT_RECORDS, 0L) }, "NaN")} </td>
+      <td> {withNoProgress(query, { query.lastProgress.batchId }, "NaN")} </td>
+      {details(withNoProgress(query, { query.exception.map(_.message).getOrElse("-") }, "-"))}
+    </tr>
+  }
+
+  private def generateStreamingQueryTable(request: HttpServletRequest): Seq[Node] = {
+    val (activeQueries, inactiveQueries) = store.allStreamQueries.partition(_._1.isActive)
+    val activeQueryTables = if (activeQueries.nonEmpty) {
+      val headerRow = Seq(
+        "Query Name", "Status", "Id", "Run ID", "Submit Time", "Duration", "Avg Input /sec",
+        "Avg Process /sec", "Total Input Rows", "Last Batch ID", "Error")
```

Comment:
It seems that only inactiveQueries need "Error" row, cause for the active queries, error row always empty.

Reply:
fixed